pycqed/analysis/analysis_toolbox.py

```diff
@@ -118,47 +118,64 @@ def return_last_n_timestamps(n, contains=''):
 
 
 def latest_data(contains='', older_than=None, newer_than=None, or_equal=False,
-                return_timestamp=False, raise_exc=True,
-                folder=None, return_all=False):
-    '''
-    finds the latest taken data with <contains> in its name.
-    returns the full path of the data directory.
-
-    if older_than is not None, then the latest data that fits and that
-    is older than the date given by the timestamp older_than is returned.
-    if newer_than is not None, than the latest data that fits and that
-    is newer than the date given by the timestamp newer_than is returned
-
-    If no fitting data is found, an exception is raised.
-    Except when you specifically ask not to to
-    this in: raise_exc = False, then a 'False' is returned.
-    return_all = True: returns all the folders that satisfy
-        the requirements (Cristian)
-    '''
-    if (folder is None):
+                return_timestamp=False, return_path=True, raise_exc=True,
+                folder=None, list_timestamps=False, return_all=False):
+    """
+        Finds the latest taken data with <contains> in its name.
+        Returns the full path of the data directory and/or the timestamp.
+        If no fitting data is found, an exception is raised (unless raise_exc=False).
+
+        Args:
+            contains: filter for the labels. Any label that does not include
+                this string will be ignored. (default: no filter)
+            older_than: if this is not None, the latest data that fits and that is
+                older than the date given by the timestamp older_than is returned.
+            newer_than: if this is not None, the latest data that fits and that is
+                newer than the date given by the timestamp newer_than is returned.
+            or_equal: switches "older/newer than" comparisons to  "older/newer than
+                or equal" comparisons to (default: False)
+            return_timestamp: return the timestamp(s) as first return value
+                (default: False)
+            return_path: return the path(s) as the only or the second first return
+                value (default: True)
+            raise_exc: Return False instead of raising an execption if no data is
+                found. (default: False)
+            folder: search directory (default: the stored datadir)
+            list_timestamps: If set to an integer n, a list of the n latest matches
```

Comment:
For nicer code, I would suggest to rename this parameter to something like "num_ts", "n_ts", or just "n". In any case something that alludes to the fact that this parameter specified latest n timestamps. list_timestamps suggests that a list should be passed or that this parameter is a boolean flag. And then I would find it nicer to have its default value to 1 for the latest timestamp, instead of False.

Reply:
Calling it num_ts would also be fine for me. The reason for the default value is compatibility with old code, which will expect a single path/timestamp as return value and not a single-entry list. On the other hand, new code that uses num_ts will expect a list as return value and it is more robust to then return a single-entry list in case of num_ts=1 in order to have a unified return format. That is why I thought the distinction between n_ts not provided and n_ts=1 might be useful. What do you think? Of course, I could use None instead of False to encode the default behavior.

Reply:
I would personally be ok with only the unified return format of lists. If you expect single answer, you can always take the first element of the list. In anycase, I think "n" or could make more sense as if I understood correctly, it's not only timestamps but also filepaths so the "list_timestamps" is a bit a misnomer, right?